codecparsers/bitReader_unittest.cpp

```diff
@@ -48,12 +48,12 @@ BITREADER_TEST(NoOverflow)
     std::vector<uint8_t>& data = largeData();
     const uint32_t nBytes = data.size();
 
-    ASSERT_EQ(0, nBytes % (1024 * 1024 * 512));
+    ASSERT_EQ(0u, nBytes % (1024 * 1024 * 512));
```

Comment:
What version of GCC?  When you compile, do you see `-Wall -Werror` compiler flags in the compiler command line output?  If not, are you're setting C/CXXFLAGS in your environment? Environment settings will override the project settings (see #382 bug for more details on that).  So either add `-Wall -Werror` to your env C/CXXFLAGS or unset them.  You can also try adding `-Wsign-compare`.

Otherwise, older GCC might not be strict or smart enough to catch these with `-Wall -Werror` and/or `-Wsign-compare`.  If we are diligently validating and testing patches on all Ubuntu LTS releases (precise, trusty, and xenial - i.e. the "compatible" platforms) before accepting them (as we ought to), then breaking others shouldn't be too problematic.  If a patch gets in that breaks a newer GCC compile, then they are easy to fix.  These things just come with the territory of tools and software evolution.  I don't think it's critical to hold up this patchset for this concern.